visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts

```diff
@@ -26,23 +26,34 @@ export class FileChooserController {
 	public onImportNewFiles(element) {
 		this.$scope.$apply(() => {
 			this.storeService.dispatch(resetFiles())
+			let content
+
 			for (const file of element.files) {
+				const isCompressed = file.name.endsWith(".gz")
 				const reader = new FileReader()
 				reader.onloadstart = () => {
 					this.storeService.dispatch(setIsLoadingFile(true))
 				}
 				reader.onload = event => {
-					this.setNewData(file.name, (<any>event.target).result)
+					if (isCompressed) {
+						const zlib = require("zlib")
+
+						content = zlib.unzipSync(Buffer.from((<any>event.target).result))
```

Comment:
The sync function is going to slow down the process if there's more than a single file loaded. The reason is that it's a blocking operation. So no other task is able to run until this is done. Do you know if there are frequently multiple files?